modules/engine-prototype/src/main/scala/engine/package.scala

```diff
@@ -0,0 +1,46 @@
+import scala.concurrent.ExecutionContext
+
+import cats.Apply
+import cats.implicits._
+import cats.effect.Effect
+
+import fs2.Stream
+import fs2.async
+import fs2.async.mutable.Queue
+
+package object engine {
+
+  def run[F[_]](in: Queue[F, Event])(st0: Sequence.State)(implicit F: Effect[F], ec: ExecutionContext): Stream[F, Sequence.State] =
+    Sequence.State.Mutable.stream[F](st0: Sequence.State)(m => forever(reader(in)(m)))
```

Comment:
This is the very top level entry point to the engine. You pass a initial State a queue from where to read input user events and you obtain a Stream of States. This should be the only function consumer code should depend on.